compiler/rustc_attr/src/builtin.rs

```diff
@@ -555,16 +574,21 @@ pub fn eval_condition(
                     return false;
                 }
             };
-            let min_version = match Version::parse(&min_version.as_str()) {
+            let min_version = match parse_version(&min_version.as_str(), false) {
                 Some(ver) => ver,
                 None => {
-                    sess.span_diagnostic.struct_span_err(*span, "invalid version literal").emit();
+                    sess.span_diagnostic
+                        .struct_span_warn(
+                            *span,
+                            "ignoring cfg(version) due to invalid version literal",
+                        )
+                        .emit();
```

Comment:
We are not ignoring it, we return `false`, it means the item `#[cfg(version("garbage"))] ITEM` will be removed.
Returning `true` would also not be an ignore, any value returned from a boolean predicate can do something (e.g. `#[cfg(not(version("garbage")))] ITEM` will be removed if we return `true`).

Also hard-coded warnings should not be used, perhaps the `UNUSED_ATTRIBUTES` lint can be used here if really necessary.

I'd suggest to keep this an error to avoid these questions.
If versioning scheme changes one day, it will likely be planned ahead, so we'll get a long enough range of releases using the old versioning, but supporting the new versioning in `cfg(version)`.

Reply:
On the other hand, in case of versioning scheme changes `version("unknown")` means that we are seeing a version from the future and therefore should return `false` because we are not in the future. But that's still not ignoring.

Reply:
Meh, this can use a hard-coded warning, it just shouldn't say "ignored".
We can always make a new lint for this when it becomes necessary.

Reply:
> On the other hand, in case of versioning scheme changes version("unknown") means that we are seeing a version from the future and therefore should return false because we are not in the future. But that's still not ignoring.

That was my idea :). Do you know about better alternatives to the "ignored" term?

Reply:
> If versioning scheme changes one day, it will likely be planned ahead, so we'll get a long enough range of releases using the old versioning, but supporting the new versioning in cfg(version).

That would be unfortunate as the entire point of the `cfg(version)` feature is to allow code to be forward compatible.